File: mishkaandgame.java
import java.util.Scanner;

public class mishkaandgame{

	static Scanner in = new Scanner(System.in);

	public static void main(String[]args){
	
		int m = 0;
		int c = 0;

		int n = in.nextInt();

		for(int i = 0; i < n; i++){
			int m_dice= in.nextInt();
			int c_dice= in.nextInt();

			if(m_dice > c_dice){
				m++;
			}
			
			if(c_dice > m_dice){
				c++;
			}
		}
		
		if(m > c){
			System.out.println("Mishka");
		
		}
		
		if(c > m){
			System.out.println("Chris");
		
		}
		
		if(c == m){
			System.out.println("Friendship is magic!^^");
		
		}
	}
}

File: mishkaandgame.py
n = int(input())
m = 0
c = 0

for i in range(n):
    m_dice, c_dice = map(int, input().split())

    if m_dice > c_dice:
        m+=1
    
    if c_dice > m_dice:
        c+=1

if m > c:
    print("Mishka")

if c > m:
    print("Chris")

if c == m:
    print("Friendship is magic!^^")

